precompiles/distribution/distribution.go

```diff
@@ -246,3 +259,89 @@ func (p PrecompileExecutor) accAddressFromArg(ctx sdk.Context, arg interface{})
 	}
 	return seiAddr, nil
 }
+
+type Coin struct {
+	Amount   *big.Int
+	Denom    string
+	Decimals *big.Int
+}
+
+type Reward struct {
+	ValidatorAddress string
+	Coins            []Coin
+}
+
+type Rewards struct {
+	Rewards []Reward
+	Total   []Coin
+}
+
+func (p PrecompileExecutor) rewards(ctx sdk.Context, method *abi.Method, args []interface{}) (ret []byte, remainingGas uint64, rerr error) {
+	defer func() {
+		if err := recover(); err != nil {
```

Comment:
do we expect any panic in the calls below? If not we should probably add a log here as well